### PR TITLE
rebrand mariner to azl in containerized-rpmbuild

### PR DESCRIPTION
**Pull request by neha170**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [X] The toolchain has been rebuilt successfully (or no changes were made to it)
- [X] The toolchain/worker package manifests are up-to-date
- [X] Any updated packages successfully build (or no packages were changed)
- [X] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [X] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [X] All package sources are available
- [X] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [X] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [X] All source files have up-to-date hashes in the `*.signatures.json` files
- [X] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [X] Documentation has been updated to match any changes to the build system
- [X] Ready to merge

---

###### Summary 

Rename occurrences of 'mariner' to 'azl' in containerized-rpmbuild codebase

###### Change Log  

- Rename occurrences of 'mariner' to 'azl' in containerized-rpmbuild codebase

###### Does this affect the toolchain?  

**YES/NO**

###### Test Methodology

- Locally built and ran containerized-rpmbuild for 3.0-dev

### CODE REVIEW

File: toolkit/scripts/containerized-build/create_container_build.sh [hunk 1]
```diff
@@ -155,11 +155,11 @@ repo_branch=$(git -C ${repo_path} rev-parse --abbrev-ref HEAD)
 # Generate text based on mode (Use figlet to generate splash text once available on Azure Linux)
 if [[ "${mode}" == "build" ]]; then
     echo -e "\033[31m -----------------------------------------------------------------------------------------\033[0m" > ${tmp_dir}/splash.txt
-    echo -e "\033[31m ----------------------------------- MARINER BUILDER ! ----------------------------------- \033[0m" >> ${tmp_dir}/splash.txt
+    echo -e "\033[31m --------------------------------- AZURE LINUX BUILDER ! -------------------------------- \033[0m" >> ${tmp_dir}/splash.txt
```
Comment (tobiasb-ms) [inline toolkit/scripts/containerized-build/create_container_build.sh:158]: supernit: Looks like you're missing a `-` at the end of this string, if we want to match up with the others near it

```suggestion
    echo -e "\033[31m --------------------------------- AZURE LINUX BUILDER ! --------------------------------- \033[0m" >> ${tmp_dir}/splash.txt
```

File: toolkit/scripts/containerized-build/create_container_build.sh [hunk 2]
```diff
@@ -155,11 +155,11 @@ repo_branch=$(git -C ${repo_path} rev-parse --abbrev-ref HEAD)
...
     echo -e "\033[31m -----------------------------------------------------------------------------------------\033[0m" >> ${tmp_dir}/splash.txt
 else
     echo -e "\033[31m -----------------------------------------------------------------------------------------\033[0m" > ${tmp_dir}/splash.txt
-    echo -e "\033[31m ----------------------------------- MARINER TESTER ! ------------------------------------ \033[0m" >> ${tmp_dir}/splash.txt
+    echo -e "\033[31m --------------------------------- AZURE LINUX TESTER ! --------------------------------- \033[0m" >> ${tmp_dir}/splash.txt
```
Comment (tobiasb-ms) [inline toolkit/scripts/containerized-build/create_container_build.sh:162]: supernit: Looks like you're missing a `-` at the end of this string, if we want to match up with the others near it

```suggestion
    echo -e "\033[31m --------------------------------- AZURE LINUX TESTER ! ---------------------------------- \033[0m" >> ${tmp_dir}/splash.txt
```

File: toolkit/scripts/containerized-build/create_container_build.sh [hunk 3]
```diff
@@ -228,25 +228,25 @@ cp resources/setup_functions.sh $tmp_dir/setup_functions.sh
...
+dockerfile="${script_dir}/resources/azl.Dockerfile"
 
 # TODO: Remove test mode when image is available for 3.0
 if [[ "${mode}" == "build" || "${mode}" == "test" ]]; then # Configure base image
     echo "Importing chroot into docker..."
     chroot_file="$BUILD_DIR/worker/worker_chroot.tar.gz"
     if [[ ! -f "${chroot_file}" ]]; then build_worker_chroot; fi
     chroot_hash=$(sha256sum "${chroot_file}" | cut -d' ' -f1)
-    container_img="mcr.microsoft.com/cbl-mariner/containerized-rpmbuild:${version}"
+    container_img="azl/containerized-rpmbuild:${version}"
```
Comment (tobiasb-ms) [inline toolkit/scripts/containerized-build/create_container_build.sh:249]: Are we sure this is right? I'll believe you if it is, but I'm surprised it isn't `mcr.microsoft.com/azl`.

Comment (neha170) [reply toolkit/scripts/containerized-build/create_container_build.sh:249]: I'm not sure either. If we are tagging images locally, with no intention of publishing them, does prefix `mcr.microsoft.com` matter? What is the protocol to follow?
We still download from `mcr.microsoft.com/cbl-mariner/..` if not building the image locally.

Comment (tobiasb-ms) [reply toolkit/scripts/containerized-build/create_container_build.sh:249]: Based on this: https://github.com/microsoft/azurelinux/blob/28d18aeeb900f1eba849921141d817fdf1e57918/.pipelines/containerSourceData/scripts/BuildGoldenContainer.sh#L51C1-L56C64, I think we may want something like:
```
mcr.microsoft.com/azurelinux/containerized-rpmbuild:3.0
```
But verify.

Comment (neha170) [reply toolkit/scripts/containerized-build/create_container_build.sh:249]: Cool, let's do that so we have consistency

**Review by tobiasb-ms — APPROVED** [review]

File: toolkit/scripts/containerized-build/create_container_build.sh [hunk 4]
```diff
@@ -266,14 +266,14 @@ fi
 
 # ================== Launch Container ==================
 echo "Checking if build env is up-to-date..."
-docker_image_tag="mcr.microsoft.com/cbl-mariner/${USER}-containerized-rpmbuild:${version}"
+docker_image_tag="mcr.microsoft.com/azurelinux/${USER}-containerized-rpmbuild:${version}"
```
Comment (PawelWMS) [inline toolkit/scripts/containerized-build/create_container_build.sh:269]: Does this already work?

Comment (PawelWMS) [reply toolkit/scripts/containerized-build/create_container_build.sh:269]: Wait, I'm just dumb.

**Review by PawelWMS — APPROVED** [review]

**Pull request merged**